### PR TITLE
Add some tracing to wasm-emscripten-finalize

### PR DESCRIPTION
**Pull request by sbc100**
Also fix, but in splitting the names of the trace channels.  Obviously
I can't write string.split correctly in C first time around.

### CODE REVIEW

File: src/tools/wasm-emscripten-finalize.cpp [hunk 1]
```diff
@@ -175,10 +178,9 @@ int main(int argc, const char* argv[]) {
 
   options.applyFeatures(wasm);
 
-  if (options.debug) {
-    std::cerr << "Module before:\n";
-    WasmPrinter::printModule(&wasm, std::cerr);
-  }
+  // Useful for debugging
```
Comment (jgravelle-google) [inline src/tools/wasm-emscripten-finalize.cpp]: It feels weird to go from a runtime check to a compile-time thing here. Maybe add a new runtime option here? `--dump-module`?
I do agree that I usually don't want a pre-module dump, so I do support decoupling debug from dump here.

Also we should update the "Module after" dump to BYN_TRACE + BYN_DEBUG too? (Line293, can't comment it directly)

File: src/wasm/wasm-emscripten.cpp [hunk 2]
```diff
@@ -1275,12 +1287,15 @@ void EmscriptenGlueGenerator::exportWasiStart() {
   // If main exists, export a function to call it per the wasi standard.
   Name main = "main";
   if (!wasm.getFunctionOrNull(main)) {
+    BYN_TRACE("exportWasiStart: main not found\n");
```
Comment (jgravelle-google) [inline src/wasm/wasm-emscripten.cpp]: Not related to this change per se:

Do we have any BYN_TRACEs that don't end with "\n" (or std::endl)? Should we make BYN_TRACE always append a newline?

File: src/wasm/wasm-emscripten.cpp [hunk 3]
```diff
@@ -229,6 +237,7 @@ ensureFunctionImport(Module* module, Name name, std::string sig) {
 // Here we internalize all such wasm globals and generte code that sets their
 // value based on the result of call `g$foo` and `fp$bar` functions at runtime.
 Function* EmscriptenGlueGenerator::generateAssignGOTEntriesFunction() {
+  BYN_TRACE("generateAssignGOTEntriesFunction\n");
```
Comment (kripken) [inline src/wasm/wasm-emscripten.cpp]: unrelated to this PR, maybe we could always emit a newline in `BYN_TRACE` so every command doesn't need to?

Comment (kripken) [reply src/wasm/wasm-emscripten.cpp]: Ah I see there is another comment on just this :)

File: src/wasm/wasm-emscripten.cpp (hunk 2 above)
Comment (sbc100) [reply src/wasm/wasm-emscripten.cpp]: I think I'll leave that for a separate PR and/or discussion.

File: src/tools/wasm-emscripten-finalize.cpp (hunk 1 above)
Comment (sbc100) [reply src/tools/wasm-emscripten-finalize.cpp]: Done

**Pull request merged**